### PR TITLE
[MI-2303] Add button to connect to ServiceNow account on error panel

### PR DESCRIPTION
**Pull request by ayusht2810**
image

image

image

image

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 1]
```diff
@@ -272,7 +274,9 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
     // Returns heading for the result panel
     const getResultPanelHeader = useCallback(() => {
         if (apiError && apiResponseValid) {
-            return apiError.message;
+            return apiError.id === Constants.ApiErrorIdNotConnected || apiError.id === Constants.ApiErrorIdRefreshTokenExpired ?
+                Utils.getContentForResultPanelIfDisconnected(apiError.message, hideModal) :
```
Comment (avas27JTG) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:278]:
```suggestion
                Utils.getContentForResultPanelWhenDisconnected(apiError.message, hideModal) :
```

File: webapp/src/containers/shareRecords/index.tsx [hunk 2]
```diff
@@ -135,6 +137,17 @@ const ShareRecords = () => {
...
 
+    // Returns heading for the result panel
+    const getResultPanelHeader = () => {
+        if (apiError) {
+            return apiError.id === Constants.ApiErrorIdNotConnected || apiError.id === Constants.ApiErrorIdRefreshTokenExpired ?
+                Utils.getContentForResultPanelIfDisconnected(apiError.message, hideModal) :
+                apiError.message;
+        }
+
+        return Constants.RecordSharedMsg;
```
Comment (avas27JTG) [inline webapp/src/containers/shareRecords/index.tsx:148]:
```suggestion
        return apiError ? (
            apiError.id === Constants.ApiErrorIdNotConnected || apiError.id === Constants.ApiErrorIdRefreshTokenExpired ?
                Utils.getContentForResultPanelIfDisconnected(apiError.message, hideModal) :
                apiError.message;
        )
       : Constants.RecordSharedMsg;
```

see if this works, I guess nested ternary not allowed by linter

File: webapp/src/containers/updateState/index.tsx [hunk 3]
```diff
@@ -87,15 +89,26 @@ const UpdateState = () => {
...
+    // Returns heading for the result panel
+    const getResultPanelHeader = () => {
+        if (apiError) {
+            return apiError.id === Constants.ApiErrorIdNotConnected || apiError.id === Constants.ApiErrorIdRefreshTokenExpired ?
+                Utils.getContentForResultPanelIfDisconnected(apiError.message, hideModal) :
+                apiError.message;
+        }
+
+        return Constants.StateUpdatedMsg;
+    };
```
Comment (avas27JTG) [inline webapp/src/containers/updateState/index.tsx:110]: we should create util for this and pass this `return Constants.StateUpdatedMsg` as an argument

File: webapp/src/containers/shareRecords/index.tsx (hunk 2 above)
Comment (manojmalik20) [reply webapp/src/containers/shareRecords/index.tsx:148]: Yes, it's not allowed. This doesn't work.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**